pyccl/halos/halo_model.py

```diff
@@ -530,7 +552,7 @@ def halomod_power_spectrum(cosmo, hmc, k, a, prof,
         `k` and `a` respectively. If `k` or `a` are scalars, the
         corresponding dimension will be squeezed out on output.
     """
-    a_use = np.atleast_1d(a)
+    a_use = np.atleast_1d(a).astype(float)
```

Comment:
Why astype(float) here but not for `k_use`?

Reply:
If you pass an integer scale factor (i.e. 1) the code breaks, so it was a quick patch. Don't think too much of it, I've taken care of this kind of error in #891 .